Modules/_io/iobase.c

```diff
@@ -447,7 +452,8 @@ _PyIOBase_check_writable(PyObject *self, PyObject *args)
         return NULL;
     if (res != Py_True) {
         Py_CLEAR(res);
-        iobase_unsupported("File or stream is not writable.");
+        _PyIO_State *state = find_io_state_by_def(Py_TYPE(self));
```

Comment:
`find_io_state_by_def` cannot be used here, you need to pass down the state from the object itself.

Reply:
Well observed. The code above also confirms this.

Reply:
Hm, my current fix won't work, unfortunately.

Reply:
You need to pass down the state from the calling function here, `self` can be any object so you cannot get the module state from it. 0cdd1eae3d33dc0fcf2ee25c395a81b262876737 also needs to be reverted.

Reply:
Sure, we can pass state as a parameter to the `_PyIOBase_check_` functions for internal `_io` use, but we'll need to remove them from `iobase_methods` as the new argument spec will be incompatible with `METH_NOARGS`.